frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java

```diff
@@ -23,20 +30,42 @@ public boolean isLoggingAcceptable() {
     return !isUpdateInProgressOnUIThread;
   }
 
+  protected abstract boolean isSideEffecting();
```

Comment:
I am wondering - maybe true as default?

Reply:
Huh I'd rather force the developer to stop and think what's the correct choice (and about its consequences)